### PR TITLE
Lots here, States fixed, Ranged State waiting on Projectile Class to be finished

### PR DESCRIPTION
**Pull request by Jeff-Moreau**

### CODE REVIEW

File: Source/SteveSloth/Private/MyGenericEnemyChaseState.cpp [hunk 1]
```diff
@@ -24,9 +25,23 @@ void UMyGenericEnemyChaseState::ExitState()
...
+
+	FVector myLocation = Myself->GetActorLocation();
+	FVector steveLocation = Player->GetActorLocation();
+
+	FVector directionToTravel = (steveLocation - myLocation).GetSafeNormal();
+
+	FVector newDirection = myLocation + directionToTravel * Myself->GetMovementSpeed() * deltaTime;
+
+	FString location = steveLocation.ToString();
+	UE_LOG(LogTemp, Warning, TEXT("im chasing steve: %s"), *location);
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyChaseState.cpp:37]: remove testing stuff

File: Source/SteveSloth/Private/MyGenericEnemyDieState.cpp [hunk 2]
```diff
@@ -24,6 +22,7 @@ void UMyGenericEnemyDieState::ExitState()
 
 void UMyGenericEnemyDieState::UpdateState(float deltaTime)
 {
+	UE_LOG(LogTemp, Warning, TEXT("im dead"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyDieState.cpp:25]: remove testing

File: Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp [hunk 3]
```diff
@@ -24,6 +22,7 @@ void UMyGenericEnemyIdleState::ExitState()
 
 void UMyGenericEnemyIdleState::UpdateState(float deltaTime)
 {
+	UE_LOG(LogTemp, Warning, TEXT("im idle"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyIdleState.cpp:25]: remove testing stuff

File: Source/SteveSloth/Private/MyGenericEnemyPatrolState.cpp [hunk 4]
```diff
@@ -14,16 +14,24 @@ void UMyGenericEnemyPatrolState::EnterState()
...
 void UMyGenericEnemyPatrolState::ExitState()
 {
-
+	
 }
 
 void UMyGenericEnemyPatrolState::UpdateState(float deltaTime)
 {
+	FVector currentSpot = Myself->GetStartingLocation().GetLocation();
+	FVector locationToGo = FMath::VInterpConstantTo(currentSpot, PatrolSpot, deltaTime, 100.f);
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyPatrolState.cpp:32]: should the 100f based on move speed? (not 100% about this one)

**Review by Saretzky — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Private/MyGenericEnemyPatrolState.cpp (hunk 4 above)
Comment (Jeff-Moreau) [reply Source/SteveSloth/Private/MyGenericEnemyPatrolState.cpp:32]: this one is based off of getting delta time to 1 instead of 1000

**Review by Saretzky — APPROVED** [review]
looks good

**Review by CrazyTobi — APPROVED** [review]
Looks good

**Pull request merged**